rfcs/0128-redeployable-clients.md

```diff
@@ -0,0 +1,393 @@
+generate the URL paths, rather than requiring the consumer of this library to
+use taskcluster-lib-urls. This keeps the involvement of taskcluster-lib-urls as
+high up in the stack as possible, which makes the lower parts of the stack more
+generic/flexible with fewer concerns.
+
+There are several benefits of including the fully qualified URLs here, rather
+than providing only the version number (`v1`), the type (`api` / `exchanges`)
+and the service name (`queue`, `auth`, ...):
+
+* teams that deploy their own taskcluster environments, are able to include
+  additional references for programming interfaces not covered by the core
+  platform (for example, maybe it is desired to provide APIs for talking with
+  databases, other messaging buses, monitoring tools, ...)
+* no assumption is made that additional APIs are uniquely identifiable with a
+  version string and a name (so long as you publish an api reference
+  *somewhere* you can include it)
+* if client generators are not able to understand a reference (because it
+  implements a schema they are not familiar with) they can ignore it; perhaps
+  the docs site will be able to display it, or something else will be consuming
+  the reference
+* nothing in this specification needs to change if new reference formats are
+  introduced
+* no redundancy in the data - if the API manifest provided information that is
+  already in the API reference itself, potentially the data may not concur
+
+This is a "keep it simple" approach that allows other parts of the system to
+adapt without this part being affected (i.e. reduces brittleness). The API
+manifest simply says "these are the APIs I declare, here is where you can fetch
+their references, and they are self-describing, so go ask them". It doesn't
+burn in any concerns about URL path building, or the types of reference we
+support.
+
+## Changes to HTTP references format
+
+The `api-reference.json` document currently served
+http://schemas.taskcluster.net/base/v1/api-reference.json from should be served
+from `<TASKCLUSTER_ROOT_URL>/schemas/base/v1/http-reference.json`.
+
+* All `entries[*].input` and `entries[*].output` properties should be URLs
+  relative to `<taskclusterRootURL>/schemas/<serviceName>`
+
+(This has already been implemented across most services)
+
+## Changes to AMQP references format
+
+The `api-reference.json` document currently served
+http://schemas.taskcluster.net/base/v1/api-reference.json from should be served
+from `<TASKCLUSTER_ROOT_URL>/schemas/base/v1/amqp-reference.json`.
+
+* All `entries[*].schema` properties should be URLs relative to
+  `<taskclusterRootURL>/schemas/<serviceName>`
+
+(This has already been implemented across most services)
+
+## Changes to publication of API references and schemas
+
+The implementation must serve the described resources under the given URLs set
+out in this document. The author is not concerned with how a service declares
```

Comment:
Let's omit from the RFC, then..

Reply:
I think it's useful information to clearly mark what is and isn't in scope.